NullPointerException on reading auth.attemptedUsername in terms template

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

authentication

### Describe the bug

Hello,

I am trying to read the auth.attemptedUsername in terms template and I get a NullPointerException  
```html
    <label id="kc-terms-attempted-username">${auth.attemptedUsername}</label>
````
https://github.com/keycloak/keycloak/blob/c4255e73018bb3dd0351cdc6e61b0767ce69c7b3/services/src/main/java/org/keycloak/forms/login/freemarker/model/AuthenticationContextBean.java#L63

```
Caused by: freemarker.core._TemplateModelException: An error has occurred when reading existing sub-variable "attemptedUsername"; see cause exception! The type of the containing value was: extended_hash+string (org.keycloak.forms.login.freemarker.model.AuthenticationContextBean wrapped into f.e.b.StringModel)
----
FTL stack trace ("~" means nesting-related):
        - Failed at: ${auth.attemptedUsername}  [in template "terms.ftl" at line 3, column 45]
        ~ Reached through: #nested "header"  [in template "template.ftl" in macro "registrationLayout" at line 81, column 24]
        ~ Reached through: @layout.registrationLayout displayMes...  [in template "terms.ftl" at line 2, column 1]
----
        at freemarker.ext.beans.BeanModel.get(BeanModel.java:185)
        at freemarker.core.Dot._eval(Dot.java:43)
        at freemarker.core.Expression.eval(Expression.java:101)
        at freemarker.core.DollarVariable.calculateInterpolatedStringOrMarkup(DollarVariable.java:100)
        at freemarker.core.DollarVariable.accept(DollarVariable.java:63)
        at freemarker.core.Environment.visit(Environment.java:383)
        at freemarker.core.Environment.invokeNestedContent(Environment.java:633)
        at freemarker.core.BodyInstruction.accept(BodyInstruction.java:60)
        at freemarker.core.Environment.visit(Environment.java:347)
        at freemarker.core.Environment.visit(Environment.java:353)
        at freemarker.core.Environment.visit(Environment.java:389)
        at freemarker.core.Environment.invokeMacroOrFunctionCommonPart(Environment.java:889)
        at freemarker.core.Environment.invokeMacro(Environment.java:825)
        at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
        at freemarker.core.Environment.visit(Environment.java:347)
        at freemarker.core.Environment.visit(Environment.java:353)
        at freemarker.core.Environment.process(Environment.java:326)
        at freemarker.template.Template.process(Template.java:383)
        at org.keycloak.theme.freemarker.DefaultFreeMarkerProvider.processTemplate(DefaultFreeMarkerProvider.java:49)
        ... 67 more
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at freemarker.ext.beans.BeansWrapper.invokeMethod(BeansWrapper.java:1552)
        at freemarker.ext.beans.BeanModel.invokeThroughDescriptor(BeanModel.java:233)
        at freemarker.ext.beans.BeanModel.get(BeanModel.java:152)
        ... 85 more
Caused by: java.lang.NullPointerException
        at org.keycloak.forms.login.freemarker.model.AuthenticationContextBean.getAttemptedUsername(AuthenticationContextBean.java:63)
        ... 92 more
```


### Version

20.0.1

### Expected behavior

To return the attemptedUsername

### Actual behavior

Throws an error

### How to Reproduce?

1. Add the HTML template snippet in the terms.ftl template
2. Sign in with a user for the first time

### Anything else?

_No response_

--- a/services/src/main/java/org/keycloak/forms/login/freemarker/model/AuthenticationContextBean.java
+++ b/services/src/main/java/org/keycloak/forms/login/freemarker/model/AuthenticationContextBean.java
@@ -60,6 +60,10 @@ public class AuthenticationContextBean {
     // NOTE: This is called "attemptedUsername" as we won't necessarily display the username of the user, but the "attempted username", which he
     // used on the login screen (which could be eventually email or something else)
     public String getAttemptedUsername() {
+        if (context == null) {
+            return null;
+        }
+
         String username = context.getAuthenticationSession().getAuthNote(AbstractUsernameFormAuthenticator.ATTEMPTED_USERNAME);
 
         // Fallback to real username of the user just if attemptedUsername doesn't exist

